src/is-of-type.ts

```diff
@@ -0,0 +1,99 @@
+import { Action, AnyAction } from './create-action'
+import { ActionCreator } from './create-action-creator'
+import { getType } from './get-type'
+import { castArray } from './utils'
+
+/**
+ * Non-curried function
+ * @example
+ * const increment = createActionCreator('[Counter] increment')
+ * const decrement = createActionCreator('[Counter] decrement')
+ * const reset = createActionCreator(
+ *   '[Counter] reset',
+ *   resolve => (value: number) => resolve(value)
+ * )
+ * isOfType(increment.type, increment()) //=> true
```

Comment:
When there is access to action creator it's not a good idea to use type property of it. Examples should be as follow:
- usage of `isOfType` with action creator(s)
- usage of `isOfType` with action(s)
- usage of `isOfTyp` with action type(s) (plain string(s)). In multiple action types case, you should cast the array type to `const`.
```ts
isOfType(<const>['[Counter] increment', '[Counter] decrement'])
```

Reply:
When there is access to action creator is it not a good idea to use action?
is the same  true of action? (plain object(s))

```
isOfType(<const>[{ type: '[Counter] increment' }, { type: '[Counter] decrement' }])
```

Reply:
When there is access to action creators, the most optimized way is just to use action creators for defining true cases. In the above code, it's better to just use strings.

Reply:
Casting an array type to `const` will result in a `ReadonlyArray`.
Passing `ReadonlyArray` to isOfType causes an type error.
Do i need to fix this?

<img width="400" src="https://user-images.githubusercontent.com/12913947/70383116-c5fcc300-19ab-11ea-9fe5-2b1d01fcc23c.png">

Reply:
Also, I don't need to cast to const when specifying array directly as an argument.

specifying array directly as an argument:

<img width="400" src="https://user-images.githubusercontent.com/12913947/70383441-4540c580-19b1-11ea-984c-cb051e924165.png">

specifying array variable as an argument:

<img width="400" src="https://user-images.githubusercontent.com/12913947/70383448-55f13b80-19b1-11ea-8899-e9a303d08f4f.png">

Is it all right if I comment example as follows?

```ts
isOfType(['[Counter] increment', '[Counter] decrement'])
```

Reply:
> Do i need to fix this?

Yeah, let's do it, We don't need a mutable array. Let's make type more general.

> specifying array directly as an argument.

Interesting! What is the TypeScript version in your environment?

Reply:
> Yeah, let's do it, We don't need a mutable array. Let's make type more general.

OK! I will!

> Interesting! What is the TypeScript version in your environment?

TypeScript version is 3.6.3!

Reply:
completed! https://github.com/thebrodmann/deox/pull/132

Reply:
> Is it all right if i comment example as follows?

You are right, I thought it won't work! :|
Good job @kotarella1110